src/BeamAdapter/component/model/WireSectionMaterial.h

```diff
@@ -0,0 +1,78 @@
+*                  (c) 2006 Inria, University of Lille, CNRS                  *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: see Authors.md                                                     *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <BeamAdapter/config.h>
+#include <BeamAdapter/utils/BeamSection.h>
+#include <sofa/core/objectmodel/BaseObject.h>
+
+namespace sofa::beamadapter
+{
+
+/**
+ * \class WireRestShape
+ * \brief Describe the shape functions on multiple segments
+ *  
+ *  Describe the full shape of a Wire with a given length and radius. The wire is discretized by a set of beams (given by the keyPoints and the relatives Beam density)
+ *  This component compute the beam discretization and the shape functions on multiple segments using curvilinear abscissa.
+ */
+class SOFA_BEAMADAPTER_API WireSectionMaterial : public core::objectmodel::BaseObject
+{
+public:
+    SOFA_CLASS(WireSectionMaterial, core::objectmodel::BaseObject);
+
+    /// Default Constructor
+    WireSectionMaterial();
+
+
+    void init();
+
+    /// This function gives the Young modulus and Poisson's coefficient of the beam depending on the beam position
+    void getYoungModulusAtX(float& youngModulus, float& cPoisson);
+    void getYoungModulusAtX(double& youngModulus, double& cPoisson);
+
+    /// This function gives the mass density and the BeamSection data depending on the beam position
+    void getInterpolationParam(float& _rho, float& _A, float& _Iy, float& _Iz, float& _Asy, float& _Asz, float& _J);
+    void getInterpolationParam(double& _rho, double& _A, double& _Iy, double& _Iz, double& _Asy, double& _Asz, double& _J);
```

Comment:
Why not a template on the floating point type?
or even, directly using SReal (the Data<> are defined with SReal after all)

In any case, we should avoid duplicating code

Reply:
casting parameter `Real&` into `SReal&`  is not working.
I was too lasy to create the .inl but yes I will template the component.